Create VILA-Live app (#334)
nvidia-holoscan/holohub: applications/vila_live/README.md

> # Holoscan VILA Live

Reviewer: Was this inspired by any other projects? (Jetson?) If so, could you please add a references section at the bottom of the README with link(s)?

Author: Added links

This applications demonstrates how to run [VILA 1.5](https://github.com/Efficient-Large-Model/VILA) models on live video feed with the possibility

> of changing the prompt in real time.

Reviewer: Could you please add 1-2 sentences summarizing what VILA is or does? What is its value proposition, or what distinguishes it from other HoloHub apps?

Author: Added

![Holoscan VILA Live](./screenshot.png)
Note: This demo currently uses [Llama-3-VILA1.5-8b-AWQ](https://huggingface.co/Efficient-Large-Model/Llama-3-VILA1.5-8b-AWQ), but any of the following AWQ-quantized models from the VILA 1.5 familty should work as long as the file names are changed in the [Dockerfile](./Dockerfile) and [run_vila_live.sh](./run_vila_live.sh):
- [VILA1.5-3b-AWQ](https://huggingface.co/Efficient-Large-Model/VILA1.5-3b-AWQ)
- [VILA1.5-3b-s2-AWQ](https://huggingface.co/Efficient-Large-Model/VILA1.5-3b-s2-AWQ)
- [Llama-3-VILA1.5-8b-AWQ](https://huggingface.co/Efficient-Large-Model/Llama-3-VILA1.5-8b-AWQ)
- [VILA1.5-13b-AWQ](https://huggingface.co/Efficient-Large-Model/VILA1.5-13b-AWQ)

> - [VILA1.5-40b-AWQ](https://huggingface.co/Efficient-Large-Model/VILA1.5-40b-AWQ)
> ## Setup Instructions

Reviewer: ```suggestion
- [VILA1.5-40b-AWQ](https://huggingface.co/Efficient-Large-Model/VILA1.5-40b-AWQ)

## Setup Instructions
```

Author: Added

The app defaults to using the video device at `/dev/video0`

To debug if this is the correct device download `v4l2-ctl`:
```bash
sudo apt-get install v4l-utils
```
To check for your devices run:
```bash

> v4l2-ctl --list-devices

Reviewer: What should the output of this command look like?

Author: Added output

```
Then map your desired video device to the source device in [vila_live.yaml](vila_live.yaml)

## Build and Run Instructions
From the Holohub main directory run the following command:
```bash

> ./dev_container build_and_run vila_live

Reviewer: 🚀

```

> Note: The first build will take **~1.5 hours** if you're on ARM64. This is largely due to building [Flash Attention 2](https://github.com/Dao-AILab/flash-attention) since ARM64 wheels are not distributed for this platform.

Reviewer: Could you please add a section on target platforms or link to the HoloHub README prerequisites section to indicate that x86_64 and IGX are supported?

Reviewer: Clarify wording: it looks like the Dockerfile is building the wheel locally, after which point it is available in the image

```suggestion
Note: The first build will take **~1.5 hours** if you're on ARM64. This is largely due to building [Flash Attention 2](https://github.com/Dao-AILab/flash-attention) since pre-built wheels are not distributed for ARM64 platforms.
```

Author: Added both

> Once the main LMM-based app is running, you will see a link for the app at `http://127.0.0.1:8050`

Reviewer: Is that port arbitrary? Is there a way for users to specify a different port to use?

```suggestion
Once the main LMM-based app is running, you will see a link for the app at `http://127.0.0.1:8050`.
```

Author: It is arbitrary, they can change the code if they want a different port.

- [VILA1.5-40b-AWQ](https://huggingface.co/Efficient-Large-Model/VILA1.5-40b-AWQ)

## ⚙️ Setup Instructions
The app defaults to using the video device at `/dev/video0`

> Note: You can use a USB webcam as the video source, or an MP4 video by following the instructions for the [V4L2_Camera](https://github.com/nvidia-holoscan/holoscan-sdk/tree/main/examples/v4l2_camera#use-with-v4l2-loopback-devices) example app.

To debug if this is the correct device download `v4l2-ctl`:
```bash
sudo apt-get install v4l-utils
```
To check for your devices run:
```bash
v4l2-ctl --list-devices
```
This command will output something similar to this:
```bash
NVIDIA Tegra Video Input Device (platform:tegra-camrtc-ca):
        /dev/media0

vi-output, lt6911uxc 2-0056 (platform:tegra-capture-vi:0):
        /dev/video0

Dummy video device (0x0000) (platform:v4l2loopback-000):
        /dev/video3
```
Determine your desired video device and edit the source device in [vila_live.yaml](vila_live.yaml)

## 🚀 Build and Run Instructions
From the Holohub main directory run the following command:
```bash
./dev_container build_and_run vila_live
```

> Note: The first build will take **~1.5 hours** if you're on ARM64. This is largely due to building [Flash Attention 2](https://github.com/Dao-AILab/flash-attention) since pre-built wheels are not distributed for this ARM64 platforms.

Reviewer: Nitpick:
```suggestion
Note: The first build will take **~1.5 hours** if you're on ARM64. This is largely due to building [Flash Attention 2](https://github.com/Dao-AILab/flash-attention) since pre-built wheels are not distributed for ARM64 platforms.
```
